src/core/featurechecklistmodel.cpp

```diff
@@ -530,6 +530,21 @@ bool FeatureCheckListModel::lessThan( const QModelIndex &left, const QModelIndex
     }
   }
 
+  if ( !groupField().isEmpty() )
+  {
+    const QString leftGroup = sourceModel()->data( left, FeatureListModel::GroupFieldRole ).toString();
+    const QString rightGroup = sourceModel()->data( right, FeatureListModel::GroupFieldRole ).toString();
+
+    if ( leftGroup.isEmpty() && !rightGroup.isEmpty() )
+      return true;
+
+    if ( !leftGroup.isEmpty() && rightGroup.isEmpty() )
+      return false;
+
+    if ( !groupField().isEmpty() && leftGroup != rightGroup )
+      return leftGroup < rightGroup;
+  }
```

Comment:
This needs a bit of tweaking. If the [x] display group name is set to true, this should sit _above_ the mSortCheckedFirst so the checked items sit on top _of their visible groups_.